### PR TITLE
Add Vault + Atlas post

### PR DESCRIPTION
**Pull request by sethvargo**
/cc @cbednarski @pearkes @armon

### CODE REVIEW

File: source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+This approach puts a significant burden of responsibility on the application itself. Atlas would need to be responsible for secret sprawl, key rolling, and break-glass procedures. Instead of building this functionality into Atlas directly, we use [Vault](https://vaultproject.io) combined with the [vault-rails](https://github.com/hashicorp/vault-rails) plugin. It should come as no surprise that your Atlas data is encrypted with Vault. We run Vault behind our within a private network with very strict access controls.
```
Comment (pearkes) [inline source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md]: "behind our within a private network"

File: source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+![Animation showing how data is encrypted in Atlas with Vault](/images/blog/how-atlas-uses-vault-for-managing-secrets/animated.gif)
+
+How is this different than having Atlas encrypt the data? First, we are delegating the encryption and decryption responsibility to a service that was designed for storing secrets. Vault has first-class support for audit logging, key and access revocation, well defined break-glass procedures, and will soon have support for key rolling. Vault is a specialized product for managing secrets.
+
+Second, we increase the number of systems an attacker must compromise in order to decrypt the data. If Atlas were responsible for the encryption, it would need to know the encryption key. This key would probably stored somewhere on disk or in memory. If an attacker gained access to the application, they could dump that encryption key in addition to all the encrypted database values rather easily, making decryption incredibly trivial.
```
Comment (pearkes) [inline source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md]: I don't think "trivial" is the best way to describe gaining access to a machine, I would re-word to be a bit more conservative.

**Comment by pearkes** [comment]
Awesome!

File: source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+Read on to learn more about how Atlas uses Vault to store and manage secret and sensitive information.
+
+READMORE
+
+## Background
+
+Before diving into why we chose Vault for storing secrets in Atlas, it is important to understand existing solutions and why they may not be the best for this use case. To keep the required scope of understanding limited, we will use GitHub OAuth tokens as the example, but many components of Atlas are encrypted using Vault.
+
+When you link Atlas with a GitHub account, GitHub responds with JSON payload. Among the many fields in that payload are some secret pieces of information such as your GitHub username and GitHub OAuth token. You can [read more about OAuth](http://oauth.net/2/) if you are not familiar, but the important point of understanding is that an OAuth token behaves like a password to the external service. In other words, having access to a user's GitHub OAtuh token provides the same level of access as the username and password. As such, it is incredibly important to keep this information secure.
```
Comment (armon) [inline source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md]: `OAtuh` typo

File: source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md (hunk 1 above)
Comment (armon) [inline source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md]: I'd replace "secret sprawl" with "key management"

File: source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Second, we increase the number of systems an attacker must compromise in order to decrypt the data. If Atlas were responsible for the encryption, it would need to know the encryption key. This key would probably stored somewhere on disk or in memory. If an attacker gained access to the application, they could dump that encryption key in addition to all the encrypted database values rather easily, making decryption incredibly trivial.
+
+By moving the encryption into Vault, **Atlas never knows the encryption or decryption keys**. Instead, Atlas is authorized to communicate with Vault with its token. The token provides limited access to Vault and can be revoked at any time. If there is suspicion of an intrusion, the token can be revoked or the Vault can be sealed. Thus, even if an attacker is able to compromise all of Atlas, they would still need to compromise Vault to be able to decrypt the encrypted database values. As an added bonus, each column of each database table is encrypted with a _different_ key, making the surface significantly larger to breach.
```
Comment (armon) [inline source/blog/2015-05-28-how-atlas-uses-vault-for-managing-secrets.md]: Replace "making the surface significantly larger to breach" with "making a potential compromise much more complex". You typically talk about an "attack surface" where large = bad.

**Comment by armon** [comment]
LGTM!

**Pull request merged**